providers/rfc2136.go

```diff
@@ -0,0 +1,182 @@
+package providers
+
+import (
+	"github.com/Sirupsen/logrus"
+	rdns "github.com/rancher/external-dns/dns"
+	"os"
+	"fmt"
+	"github.com/miekg/dns"
+	"strings"
+	"net"
+	"time"
+)
+
+type RFC2136Handler struct {
+	server     string
+	port       string
+	keyname    string
+	key        string
+	rootdomain string
+}
+
+/* Private functions */
+func needFQDN(domain string) string {
+	if (strings.HasSuffix(domain, ".")) {
+		return domain
+	} else {
+		return domain + "."
+	}
+}
+
+func (b RFC2136Handler) sendMessage(msg *dns.Msg) error {
+	t := new(dns.Transfer)
+
+	nameserver := net.JoinHostPort(b.server, b.port)
+	msg.SetTsig(b.keyname, "hmac-md5.sig-alg.reg.int.", 300, time.Now().Unix())
+	t.TsigSecret = map[string]string{b.keyname: b.key}
+
+	_, err := t.In(msg, nameserver)
+	t.Close()
+	return err
+}
+
+func (b RFC2136Handler) add(fqdn string, ip string, class string, ttl int) error {
+	logrus.Debugf("adding dns entry %s %s %d", fqdn, ip, ttl)
+	m := new(dns.Msg)
+	m.SetUpdate(b.rootdomain)
+	rr, err := dns.NewRR(fmt.Sprintf("%s %d %s %s", needFQDN(fqdn), ttl, class, ip))
```

Comment:
You can expect the fqdn field in `rdns.DnsRecord` objects to always be a fully-qualified name.